src/wallet/api/wallet.cpp

```diff
@@ -487,6 +490,8 @@ uint64_t WalletImpl::approximateBlockChainHeight() const
 }
 uint64_t WalletImpl::daemonBlockChainHeight() const
 {
+    if (!m_is_connected || m_is_shutting_down)
+        return 0;
```

Comment:
good point. but these lines doesn't change that behavior. It just prevents the rpc call from being made a few lines below which will throw an error and also return 0.